rest/rest-api/src/main/java/org/ow2/proactive_grid_cloud_portal/common/SchedulerRestInterface.java

```diff
@@ -292,23 +352,73 @@ boolean killJob(@HeaderParam("sessionid")
     @GET
     @Path("jobs/{jobid}/html")
     @Produces("application/json;charset=" + ENCODING)
-    String getJobHtml(@HeaderParam("sessionid")
-    String sessionId, @PathParam("jobid")
-    String jobId) throws IOException, NotConnectedRestException;
+    String getJobHtml(@HeaderParam("sessionid") String sessionId, @PathParam("jobid") String jobId)
+            throws IOException, NotConnectedRestException;
 
     /**
      * Returns a list of taskState 
      * @param sessionId a valid session id
      * @param jobId the job id
-     * @return a list of task' states of the job <code>jobId</code>
+     * @return a list of task' states of the job <code>jobId</code> and the total number
      */
     @GET
     @GZIP
     @Path("jobs/{jobid}/taskstates")
     @Produces("application/json")
-    List<TaskStateData> getJobTaskStates(@HeaderParam("sessionid")
-    String sessionId, @PathParam("jobid")
-    String jobId) throws NotConnectedRestException, UnknownJobRestException, PermissionRestException;
+    TaskStateDataPage getJobTaskStates(@HeaderParam("sessionid") String sessionId,
+            @PathParam("jobid") String jobId)
+                    throws NotConnectedRestException, UnknownJobRestException, PermissionRestException;
+
+    /**
+     * Returns a list of taskState with pagination
+     * @param sessionId a valid session id
+     * @param jobId the job id
+     * @param offset the index of the first TaskState to return
+     * @param limit the index (non inclusive) of the last TaskState to return
+     * @return a list of task' states of the job <code>jobId</code> and the total number
+     */
+    @GET
+    @GZIP
+    @Path("jobs/{jobid}/taskstates/paginated")
+    @Produces("application/json")
+    TaskStateDataPage getJobTaskStatesPaginated(@HeaderParam("sessionid") String sessionId,
+            @PathParam("jobid") String jobId, @QueryParam("offset") @DefaultValue("0") int offset,
+            @QueryParam("limit") @DefaultValue("50") int limit)
```

Comment:
limit parameter must be configurable and not hardcoded in the function definition